### DISCUSSION TITLE
tutorials/intro-hotchocolate/02-graphql-basics

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#106] [course feedback]
# tutorials/intro-hotchocolate/02-graphql-basics

Learn about the journey of a GraphQL operation and the components that make up a GraphQL server, such as the schema, resolvers and data sources.

https://www.apollographql.com/tutorials/intro-hotchocolate/02-graphql-basics

**Comment by lars-thurman** [comment]
As a .NEt developer since it was a Beta this statement is not correct.

In contrast, in C#, all types are non-nullable by default. Nullable types are denoted with a ?.

strings and object do not require a ? after then to denote they are nullable and never have.
go to your project file and disable the following or start using Visual Studio 2026

    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>

to 

    <ImplicitUsings>disable</ImplicitUsings>
    <Nullable>disable</Nullable>

**Comment by sphaxn** [comment]
In the diagrams illustrating the communication between client and server, the lettering is hard to read.  The lettering would be much more legible if the lettering were in a crisp, printed typeface instead of hand written (or pseudo hand written).

In contrast, the diagrams of the example graphs are very legible and easy to read.
